Feature/user form

scratchglory opened the pull request:
Finalizing the user form:
- created inputs
- refactored to keep DRY
- added navigation bar with home and register links

NEEDS:
- to have endpoints and talk to backend

scratchglory committed:
added form inputs, buttons, and navbar

scratchglory committed:
changed branch name

scratchglory committed:
removed password in reg form

williamschwindt commented:
Looks good so far! Address, phone number and email will have to be added because they are not nullable in our DB

scratchglory committed:
added groomer and customer forms

scratchglory committed:
edited body for posting to backend

coryortega commented on src/components/Forms/Register-Customer.js:
```diff
@@ -0,0 +1,206 @@
+      city: '',
+      state: '',
+      country: '',
+      photo_url: '',
```
Since we're initializing our state here, this would be a great spot to have 'photo_url' set to a default profile image url instead of an empty string.

coryortega commented on src/components/Forms/Register-Customer.js:
```diff
@@ -0,0 +1,206 @@
+  // must be async in order to await key function
+  async postDataGroomers() {
+    try {
+      let result = await fetch(
```
We do have Axios installed as a dependency and you can use that as an alternative to fetch(). You also wouldn't have to stringify the data, as Axios will do that

coryortega commented on src/components/Forms/Register-Customer.js:
```diff
@@ -0,0 +1,206 @@
+          },
+          // stringify the json obj payload
+          body: JSON.stringify({
+            fName: this.state.fName,
+            lName: this.state.lName,
+            email: this.state.email,
+            phone: this.state.phone,
+            address: this.state.address,
+            city: this.state.city,
+            state: this.state.state,
+            country: this.state.country,
+            zipcode: this.state.zip,
+            photo_url: this.state.photo_url,
```
Consider storing this data in an object called "user"

coryortega commented on src/components/Forms/Register-Customer.js:
```diff
@@ -0,0 +1,206 @@
+    }
+  }
+
+  componentDidMount() {
+    this.setState({
+      fName: this.state.fName,
+      lName: this.state.lName,
+      email: this.state.email,
+      phone: this.state.phone,
+      address: this.state.address,
+      city: this.state.city,
+      state: this.state.state,
+      country: this.state.country,
+      zipcode: this.state.zip,
+      photo_url: this.state.photo_url,
+      //   password: this.state.password,
+    });
+  }
```
I think this code might be redundant. Your state is getting initialized on line 6, so there isn't any need to change their values when the component mounts.

coryortega commented on src/components/Forms/Register-Customer.js:
```diff
@@ -0,0 +1,206 @@
+
+  render() {
+    //   change state based on whats coming in input
+    const { fName } = this.state;
```
Nice use of destructuring!

coryortega commented on src/components/Forms/Register.js:
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+class Register extends Component {
+  render() {
+    return (
+      <div>
+        <h2>Are you a...</h2>
+        <Link to="/groomers">Groomer</Link>
+        <Link to="/customers">Customer</Link>
+      </div>
+    );
+  }
+}
```
This is a great place to start transitioning from class components to functional components

coryortega commented on src/components/Layouts/Navbar.js:
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+class Navbar extends Component {
+  render() {
+    return (
+      <nav className="navbar">
+        <div className="container">
+          <Link className="navbar-brand" to="/">
+            Express Groomer
+          </Link>
+          <Link className="nav-link" to="/register">
+            Register
+          </Link>
+        </div>
+      </nav>
+    );
+  }
+}
+
+export default Navbar;
```
nice work on a navbar!

coryortega reviewed (changes requested):
A lotta great work here! I left some comments for some minor things and any comments that apply to "Register-Customer" also apply to "Register-Groomer". The only large thing to consider is using functional components over class components but fantastic work!

luisabellan committed:
Merge branch 'main' of github.com:Lambda-School-Labs/LabsPT12-Express_Groomer-TeamA-FE into feature/user-form

luisabellan committed:
pulled from main branch and npm update

luisabellan committed:
change placeholder text from "walk_rate" to "walk rate"

luisabellan committed:
Created e2e tests for customer and groomer registrations

luisabellan committed:
save

luisabellan committed:
save

luisabellan committed:
save

luisabellan committed:
save

luisabellan committed:
changed post request

It was pointing at /groomers, now it's pointing at /customers

luisabellan committed:
customer registration form finished

changed some variable names to what the backend was expecting, also added description

luisabellan committed:
groomer register form done

luisabellan committed:
added email to userDefault and deleted the rest of them

luisabellan committed:
fixed FormInput.js

scratchglory committed:
removed navbar

scratchglory committed:
made changes from pull request

scratchglory committed:
removed commented out data from groomer registration

scratchglory committed:
removed commented out data from groomer registration

scratchglory committed:
removed the old navbar, and unnecessary commented lines
